test/rekt/features/trigger/trigger_sink_resolution.go

```diff
@@ -0,0 +1,143 @@
+	if err := prober.ExpectYAMLEvents(eventlibrary.PathFor("events/three.ce")); err != nil {
+		panic(fmt.Errorf("can not find event files: %s", err))
+	}
+
+	// Setup Probes
+	f.Setup("install recorder", prober.ReceiverInstall("sink"))
+
+	// Setup trigger
+	f.Setup("install trigger", trigger.Install(
+		triggerName,
+		brokerName,
+		trigger.WithSubscriber(nil, "bad://uri"),
+		delivery.WithDeadLetterSink(prober.AsKReference("sink"), "")))
+
+	// Resources ready.
+	f.Setup("trigger goes ready", trigger.IsReady(triggerName))
+
+	// Install sender.
+	f.Setup("install source", prober.SenderInstall("source"))
+
+	// After we have finished sending.
+	f.Requirement("sender is finished", prober.SenderDone("source"))
+
+	// Assert events ended up where we expected.
+	f.Stable("trigger with DLS").
+		Must("accepted all events", prober.AssertSentAll("source")).
+		Must("deliver event to DLS", prober.AssertReceivedAll("source", "sink"))
+
+	return f
+}
+
+// SourceToTriggerSinkWithDLSDontUseBrokers tests to see if a Ready Trigger sends
+// failing events to it's DLS even when it's corresponding Ready Broker also have a DLS defined.
+//
+// source ---> broker --[trigger]--> bad uri
+//               |          |
+//               +--[DLS]   +--[DLS]--> sink
+//
+func SourceToTriggerSinkWithDLSDontUseBrokers(triggerName, brokerName, brokerSinkName string, prober *eventshub.EventProber) *feature.Feature {
+	triggerSinkName := "trigger-sink"
+	prober.SetTargetResource(broker.GVR(), brokerName)
+
+	f := feature.NewFeature()
+
+	lib := feature.MakeRandomK8sName("lib")
+	f.Setup("install events", eventlibrary.Install(lib))
+	f.Setup("event cache is ready", eventlibrary.IsReady(lib))
+	f.Setup("use events cache", prober.SenderEventsFromSVC(lib, "events/three.ce"))
+	if err := prober.ExpectYAMLEvents(eventlibrary.PathFor("events/three.ce")); err != nil {
+		panic(fmt.Errorf("can not find event files: %s", err))
+	}
+
+	// Setup Probes
+	f.Setup("install trigger recorder", prober.ReceiverInstall(triggerSinkName))
+	f.Setup("install brokers recorder", prober.ReceiverInstall(brokerSinkName))
+
+	// Setup topology
+	brokerConfig := append(
+		broker.WithEnvConfig(),
+		delivery.WithDeadLetterSink(prober.AsKReference(brokerSinkName), ""))
+	f.Setup("install broker with DLS", broker.Install(
+		brokerName,
+		brokerConfig...,
+	))
+
+	f.Setup("install trigger", trigger.Install(
+		triggerName,
+		brokerName,
+		trigger.WithSubscriber(nil, "bad://uri"),
+		delivery.WithDeadLetterSink(prober.AsKReference(triggerSinkName), "")))
+
+	// Resources ready.
+	f.Setup("trigger goes ready", trigger.IsReady(triggerName))
+
+	// Install events after topology is ready.
+	f.Setup("install source", prober.SenderInstall("source"))
+
+	// After we have finished sending.
+	f.Requirement("sender is finished", prober.SenderDone("source"))
+
+	// Assert events ended up where we expected.
+	f.Stable("trigger with a valid DLS ref defined in its spec").
```

Comment:
Nit, for shortening test names that end up being huge with this framework:
```suggestion
	f.Stable("trigger with a valid DLS ref").
```